File: cses/two-sum.py
from collections import defaultdict


def solve():
    _, x = map(int, input().split())
    nums = list(map(int, input().split()))

    mp = defaultdict(int)
    for i, num in enumerate(nums):
        if mp.get(x - num):
            print(i + 1, mp[x - num])
            return
        mp[num] = i + 1
    print("IMPOSSIBLE")


def main():
    solve()


if __name__ == "__main__":
    main()

File: cses/two-sum.cpp
#include <iostream>
#include <map>

#define ll long long

using namespace std;

void solve() {
    ll n, x;
    cin >> n >> x;

    map<ll, int> mp;
    for (int i = 0; i < n; ++i) {
        ll num;
        cin >> num;

        if (mp.find(x - num) != mp.end()) {
            cout << i + 1 << " " << mp[x - num] << '\n';
            return;
        }

        mp[num] = i + 1;
    }
    cout << "IMPOSSIBLE" << '\n';
}

int main() {
    solve();
    return 0;
}
